rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CreateTableResponse'
+    put:
+      tags:
+        - Catalog API
+      summary: Rename a table from its current name to a new name within the same catalog
+      description: Rename a table within the same catalog
+      operationId: renameTable
+      requestBody:
+        description: Current table identifier to rename and new table identifier to rename to
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RenameTableRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized
+        # Not using 404 as 404 is very hard to monitor.
+        # There's no way to tell by monitoring returned response codes
+        # if a client made a valid request but the table didn't exist,
+        # or if a client or server has been misconfigured or some bug exists
+        # and the client is actually calling endpoints that the server doesn't
+        # have.
+        #
+        # We can consider a different response code to use than 412.
+        '412':
+          description: Table to rename from does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchTableError'
+              example:
+                '{ error: { message: "Table does not exist", type: "NoSuchTableException", code: 41202 }'
+        '409':
+          description: The new table identifier, the to table rename to, already exists.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TableAlreadyExistsError'
+              example:
+                '{ error: { message: "Namespace already exists", type: "AlreadyExistsException", code: 40902 }'
+  /v1/namespaces/{namespace}/properties:
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - Catalog API
+      summary: Load the metadata properties for a namespace
+      operationId: loadNamespaceMetadata
+      description: Return all stored metadata properties for a given namespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application:
+              schema:
+                $ref: '#/components/schemas/GetNamespaceResponse'
+        '417':
+          description: Namespace not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchNamespaceError'
+              example:
+                '{ error: { message: "Namespace does not exist", type: "NoSuchNamespaceException", code: 41701 }'
+    put:
```

Comment:
I'd probably opt to have either POST or PUT, but not both. It is more complicated to have both with slightly different behaviors.

Reply:
+1 to either use POST or PUT. Also it seems weird to use PUT to actually delete properties. PUT is supposed to be idempotent

Reply:
I agree that having PUT and POST is a bit confusing. The reason I switched to PUT here is that a `DELETE` request isn't supposed to have a request body.

I could likely model all of this into one POST request though, where the instructions are in the body or not.

Let me do that.

Reply:
Sounds like the right thing is a POST with a body that contains both properties to set and properties to remove.

Reply:
agreed, POST definitely seems to be the right option here

Reply:
I updated this to a single POST request. It has three parameters:

- Path parameters: Namespace - URI encoded namespace
- Request body parameters:
   - toUpdate: Object containing key-value pairs to upsert
   - toRemove: Array (set) of keys to remove from the properties

It would be an invalid request to specify a key from toRemove in toUpdate as well (need to add that in).

Reply:
Added 422 (Unprocessable Entity) to indicate that there was a key in both toRemove and toUpdate.